### ISSUE TITLE
Accessibility Issue - Links in Product Cards Lack Clear Purpose (WCAG 2.4.4 Level A)

### ISSUE THREAD

**Issue by sherenemorris03** [DFE-Digital/find-information-products-services-frontend#123]
**Describe the bug**
Not in MVP
Issue 15 in Accessibility Report
The links within the product cards act as filters for the list of products below. However, this functionality is not clear from the link text alone. Screen reader users would not be aware that selecting these links will filter the product list, which may lead to confusion or missed functionality.

User story
As a screen reader user,
I want to understand the purpose of links within product cards,
So that I know what will happen when I interact with them.

**Steps to reproduce**
1. Navigate to the Manage Product Entries page.
2. Use a screen reader to explore the links within product cards.
3. Observe that the link text does not indicate its filtering function.

**Expected behaviour**
Link text or additional accessible descriptions should clearly convey the purpose and outcome of interacting with the link.

**Screenshots**
<img width="946" height="311" alt="Image" src="https://github.com/user-attachments/assets/e74d8b08-ddb3-47b2-8d11-c66aa144bbc4" />

**Additional context**
Provide some additional, hidden text for screen reader users which tells them that this will filter the list of results below. An aria-live attribute could also be used for the ‘990 products found’ text so that users are aware that the filter was successful and how many results have been returned.
